radar_parlamentar/importadores/sen.py

```diff
@@ -316,22 +317,37 @@ def progresso(self):
         sys.stdout.write('x')
         sys.stdout.flush()
 
+    '''
+    ARRRUMAR AQUI
+    '''
     def _xml_file_names(self):
         """Retorna uma lista com os caminhos dos arquivos XMLs contidos
         na pasta VOTACOES_FOLDER"""
-        files = os.listdir(VOTACOES_FOLDER)
-        xmls = [name for name in files if name.endswith('.xml.bz2')]
-        xmls = [os.path.join(VOTACOES_FOLDER, name) for name in xmls]
+        XML_URL = 'http://legis.senado.leg.br/dadosabertos/dados' + \
+            '/ListaVotacoes%d.xml'
+        ANOS_DISPONIVEIS = [2015, 2005, 2006, 2007, 2008, 2009, 2010, 2011,
+                            2012, 2013, 2014, 2016, 2017]
+
+        xmls = []
+        for ano in ANOS_DISPONIVEIS:
+            xmls.append(XML_URL % ano)
         return xmls
 
     def importar_votacoes(self):
-        """# for xml_file in
-        ['importadores/dados/senado/votacoes/ListaVotacoes2014.xml.bz2',
-        'importadores/dados/senado/votacoes/ListaVotacoes2015.xml.bz2']:
-        # facilita debug"""
-        for xml_file in self._xml_file_names():
-            logger.debug('Importando %s' % xml_file)
-            self._save_votacao_in_db(xml_file)
+        """# for xml_url in xml_list
+        http://legis.senado.leg.br/dadosabertos/dados/
+        ListaVotacoes%d.xml
+        %d é iterado em uma lista de anos entre 1991 e 2017
+         # facilita debug"""
+        try:
```

Comment:
O uso de `try/except` deve ser sempre o mais restritivo possível.

Isso significa que se a excessão que ele está pegando é gerada pela chamada `requests.get`, então o try/except deveria ficar apenas ao redor desta chamada.

Percebam que da forma como foi feita, se na minha "lista de xmls" eu tenho 10 URLs e o `get` falha na primeira (por algum problema de rede, ou porque esse arquivo específico foi tirado do ar, p.ex.) ele vai levantar excessão e não vai tentar os demais arquivos.... Por isso é importante fazer o try-except só na parte que "precisa".

Nesse nosso caso, acho que ideamente seria todo o corpo interno do `for` (as 3 linhas).